src/core/public/application/utils/get_app_info.ts

```diff
@@ -16,17 +16,19 @@ import {
 } from '../types';
 
 export function getAppInfo(app: App): PublicAppInfo {
-  const navLinkStatus =
-    app.navLinkStatus === AppNavLinkStatus.default
-      ? app.status === AppStatus.inaccessible
-        ? AppNavLinkStatus.hidden
-        : AppNavLinkStatus.visible
-      : app.navLinkStatus!;
-  const { updater$, mount, ...infos } = app;
+  const { updater$, mount, navLinkStatus = AppNavLinkStatus.default, ...infos } = app;
   return {
     ...infos,
     status: app.status!,
-    navLinkStatus,
+    navLinkStatus:
+      navLinkStatus === AppNavLinkStatus.default
+        ? app.status === AppStatus.inaccessible
+          ? AppNavLinkStatus.hidden
+          : AppNavLinkStatus.visible
+        : navLinkStatus,
+    searchable:
+      app.searchable ??
+      (navLinkStatus === AppNavLinkStatus.default || navLinkStatus === AppNavLinkStatus.visible),
```

Comment:
@pgayvallet I am not sure about that, (I am guessing in your last line of code you meant: `const computedSearchable = computedNavLinkStatus === AppNavLinkStatus.visible`)

You are right, we could compute the default `searchable` this way at the root level, but in the deepLinks we can not do that because for them the `navLinkStatus` defaults to hidden, and if any is defined, the `searchable` flag should default to true anyway.

I tried different ways to compute the `searchable` default values, and in the end, I considered the combination of `status: inaccessible` and `searchable: true` to be valid, because the first thing the global search does is filter the inaccessible status apps, so it is just the `status: inaccessible` takes precedence over the rest, and we could ignore the `status` value when we are defaulting the `searchable` flag. 

Having that in mind I could use the same condition to compute the default `searchable` value on both, the root level and the deepLinks level. Taking only the original `navLinkStatus` parameter instead of the computed one.

This is the reason why I did it this way, but I am open change it if you think the `searchable` default value should depend on the status, I think that's also okay.